src/common/lib/client/push.ts

```diff
@@ -6,14 +6,82 @@ import ErrorInfo from '../types/errorinfo';
 import PushChannelSubscription from '../types/pushchannelsubscription';
 import BaseClient from './baseclient';
 import Defaults from '../util/defaults';
+import {
+  ActivationStateMachine,
+  DeregisterCallback,
+  LocalDeviceConstructor,
+  RegisterCallback,
+} from 'plugins/push/pushactivation';
+import Platform from 'common/platform';
+import { ErrCallback } from 'common/types/utils';
 
 class Push {
   client: BaseClient;
   admin: Admin;
```

Comment:
I would be in favour of this, but it would unfortunately be a breaking change so I don't think it's worth considering for this PR, definitely worth considering for a future major release though